src/AutoRest.CSharp/Common/Input/InputTypes/InputNullableType.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace AutoRest.CSharp.Common.Input.InputTypes
+{
+    internal record InputNullableType(InputType Type): InputType("nullable")
```

Comment:
let us make this sealed because we have so many places asserting `if (type is InputNullableType)`